rusty-machine/src/lib.rs

```diff
@@ -145,8 +145,8 @@ pub mod learning {
 
             /// Return the optimized parameter using gradient optimization.
             ///
-            /// Takes in a set of starting parameters and related model data.
-            fn optimize(&self, model: &M, start: &[f64], data: &M::Data, outputs: &M::Target) -> Vec<f64>;
+            /// Takes in a set of starting parameters and related model Inputs.
+            fn optimize(&self, model: &M, start: &[f64], inputs: &M::Inputs, outputs: &M::Targets) -> Vec<f64>;
```

Comment:
`outputs` here should be called `targets` (to match the associated type).

I think this change needs to go through to the optimization algorithms themselves too.